<a href="https://colab.research.google.com/github/DeborAguiar/BreakingSilence/blob/main/notebook/BreakingSilence_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1><center>Violence Against Women and Girls</center></h1>

<center><img src="https://equineteurope.org/wp-content/uploads/2019/11/stop-violence-against-women.jpg" width="800"></img></center>



# Content

- <a href='#1'>Introduction</a>  
- <a href='#2'>Analysis preparation</a>  
- <a href='#3'>What demographics are telling us</a>   
- <a href='#4'>What the question answered reveals</a>
- <a href='#5'>Demographics and question answered combined</a>
- <a href='#6'>Median violence percent per country</a>  
- <a href='#7'>Maxim violence percent per country</a>
- <a href='#8'>Summary of findings</a>

# <a id='1'>Introduction</a>

We will analyze the data on violence on women and girls. The data is collected from 70 countries and contains demographics questions and answers as well as percent of response to 5 violence specific questions.



# <a id='2'>Analysis preparation</a>

We input the data, we glimpse the data, we look to missing data and we look to number of categories for each type of categorical data.

## Data input

In [1]:
import pandas as pd
data_df = pd.read_csv("https://github.com/DeborAguiar/BreakingSilence/blob/main/data/violence_data.csv?raw=true",
                               encoding='utf8', sep=",")

## Glimpse the data

In [2]:
data_df.head()

RecordID      Country Gender Demographics Question  \
0         1  Afghanistan      F        Marital status   
1         1  Afghanistan      F             Education   
2         1  Afghanistan      F             Education   
3         1  Afghanistan      F             Education   
4         1  Afghanistan      F        Marital status   

          Demographics Response                   Question Survey Year  Value  
0                 Never married  ... if she burns the food  01/01/2015    NaN  
1                        Higher  ... if she burns the food  01/01/2015   10.1  
2                     Secondary  ... if she burns the food  01/01/2015   13.7  
3                       Primary  ... if she burns the food  01/01/2015   13.8  
4  Widowed, divorced, separated  ... if she burns the food  01/01/2015   13.8

In [3]:
data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12600 entries, 0 to 12599
Data columns (total 8 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   RecordID               12600 non-null  int64  
 1   Country                12600 non-null  object 
 2   Gender                 12600 non-null  object 
 3   Demographics Question  12600 non-null  object 
 4   Demographics Response  12600 non-null  object 
 5   Question               12600 non-null  object 
 6   Survey Year            12600 non-null  object 
 7   Value                  11187 non-null  float64
dtypes: float64(1), int64(1), object(6)
memory usage: 787.6+ KB


In [4]:
columns = data_df.columns
for column in columns:
    print(f"{column} : {data_df[column].nunique()}")

RecordID : 420
Country : 70
Gender : 2
Demographics Question : 5
Demographics Response : 15
Question : 6
Survey Year : 18
Value : 757


# <a id='3'>What demographics are telling us</a>


Let's look first to what demographics are telling to us. There are 5 demographics questions:
* Age (15-24, 25-34, 35-49)
* Education (No Education, Primary, Secondary, Higher)
* Employement (Unemployed, Employed for cash, Employed for kind)
* Marital status (Maried or living together, Widowed, divorced, separated, Never Married)
* Residence (Rural, Urban)

We will aggregate the data on the demographic question, answer and calculate the minimm, maximum, median and average stats per each of this demographic.

In [5]:
demographics_df = data_df.groupby(["Demographics Question", "Demographics Response"])["Value"].agg(["median", "max", "min", "mean"]).reset_index()
demographics_df.columns = ["Question", "Response", "Median", "Max", "Min", "Mean"]
print("Violence % median, min, max, and mean per demographic group")
demographics_df.sort_values(["Question", "Median"])

Violence % median, min, max, and mean per demographic group


Question                      Response  Median   Max  Min       Mean
2              Age                         35-49   14.15  81.0  0.2  19.336412
1              Age                         25-34   14.45  81.5  0.1  19.703562
0              Age                         15-24   17.50  80.1  0.1  21.084169
3        Education                        Higher    4.20  74.6  0.0   8.898670
6        Education                     Secondary   13.05  76.7  0.2  17.378892
5        Education                       Primary   18.40  80.5  0.1  22.819093
4        Education                  No education   21.55  82.0  0.0  25.403125
9       Employment                    Unemployed   14.55  80.1  0.0  19.539710
7       Employment             Employed for cash   14.85  81.5  0.1  19.553804
8       Employment             Employed for kind   20.15  86.9  0.3  24.445541
10  Marital status    Married or living together   14.80  82.1  0.2  20.181003
11  Marital status                 Never married   14.90  72.1  0.0  18.704360
12  Marital status  Widowed, divorced, separated   17.15  75.9  0.0  20.683511
14       Residence                         Urban   11.05  74.4  0.1  15.895778
13       Residence                         Rural   19.00  82.1  0.2  23.294063

The data grouped on demographics shows the following:
* The most exposed age group is the 15-24, with a median of 17.50 and max of 80.1, while the absolute max is for 25-34 age group;
* Education level of girls and women is a very good predictor for violence against them: a higher education level will ensure a small median percent of 4.2%, while No education shows a 21.55% median and a 82% maximum violence percent.
* Employment factor also counts, from Unemployed to Employed for kind the median is varying from 14.55% to 20.15%. The women that work tend to be more exposed to violence.
* Marriage is slighly protecting if we look to the median, but married women are more exposed to higher level of violence. Widowed, divorced or separated have a smaller median than Married or living together group, as well as the max. But it might be also an age factor here.
* Women from Urban areas are less exposed to violence than women from Rural areas, both the median and the max values being smaller in Urban vs. Rural areas.

# <a id='4'>What the question answered reveals</a>

In [6]:
question_df = data_df.groupby(["Question"])["Value"].agg(["median", "max", "min", "mean"]).reset_index()
question_df.columns = ["Question", "Median", "Max", "Min", "Mean"]
print("Violence % median, min, max, and mean per question asked")
question_df.sort_values(["Median"])

Violence % median, min, max, and mean per question asked


Question  Median   Max  Min       Mean
2                ... if she burns the food     6.4  56.7  0.0   9.203445
5  ... if she refuses to have sex with him     9.0  68.7  0.0  13.209613
1               ... if she argues with him    15.7  76.5  0.0  18.983652
3  ... if she goes out without telling him    16.4  77.0  0.0  20.046321
4         ... if she neglects the children    20.8  75.6  0.0  23.461249
0     ... for at least one specific reason    31.0  86.9  0.0  33.217152

Generally, the median and maximum are aligned. The reasons for violence, ranged by median values are:
* if she burns the food;  
* if she refuses to have sex with him;  
* if she argues with him;  
* if she goes out without telling him;  
* if she neglects the children.  

Non specified but at least one specific reason would have the maximum median value and the absolute maximum percent.

# <a id='5'>Demographics and question answered combined</a>

In [7]:
demoq_df = data_df.groupby(["Demographics Question", "Demographics Response", "Question"])["Value"].agg(["median", "max", "min", "mean"]).reset_index()
demoq_df.columns = ["Demographics Question", "Demographics Response", "Question", "Median", "Max", "Min", "Mean"]
print("Violence % median, min, max, and mean per demographic group and question asked")
demoq_df = demoq_df.sort_values(["Demographics Question", "Demographics Response", "Median"])
demoq_df.head()

Violence % median, min, max, and mean per demographic group and question asked


Demographics Question Demographics Response  \
2                   Age                 15-24   
5                   Age                 15-24   
3                   Age                 15-24   
1                   Age                 15-24   
4                   Age                 15-24   

                                  Question  Median   Max  Min       Mean  
2                ... if she burns the food    8.50  47.7  0.1  10.098387  
5  ... if she refuses to have sex with him   10.05  59.4  0.3  13.404762  
3  ... if she goes out without telling him   18.40  67.7  0.2  21.026563  
1               ... if she argues with him   18.65  65.5  0.5  20.304032  
4         ... if she neglects the children   23.20  61.4  1.9  25.306250

In [8]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

def show_answers_per_demographic_question(question, aggregate="Median", layout_mode="Single"):
    font_size = 8
    title_font_size = 12
    if layout_mode == "Single":
        fig = go.Figure()
        font_size = 10
    else:
        fig = make_subplots(rows=2, cols=2, specs=[[{'type': 'polar'}]*2]*2)

    sel_df = demoq_df.loc[demoq_df["Demographics Question"]==question]
    answers = list(sel_df["Demographics Response"].unique())
    count = 0
    for answer in answers:
        subsel_df = sel_df.loc[sel_df["Demographics Response"] == answer]
        if layout_mode == "Single":
            fig.add_trace(go.Scatterpolar(name=answer, r=subsel_df[aggregate], theta=subsel_df['Question']))
        else:
            fig.add_trace(go.Scatterpolar(name=answer, r=subsel_df[aggregate], theta=subsel_df['Question']), count%2 + 1, int(count/2+1))
        count = count + 1
    fig.update_traces(fill='toself')
    fig.update_polars(bgcolor='white')
    fig.update_polars(angularaxis_linecolor='darkgrey')
    fig.update_polars(angularaxis_gridcolor='darkgrey')
    fig.update_polars(radialaxis_gridcolor='darkgrey')
    fig.update_polars(radialaxis_linecolor='darkgrey')
    fig.update_layout(title='Demografics Question: '+ question + ' | Statistics: ' + aggregate,font=dict(family="Courier New, monospace",size=title_font_size))
    fig.show()

In [9]:
show_answers_per_demographic_question("Age")

In [10]:
show_answers_per_demographic_question("Age", aggregate="Max")

In [11]:
show_answers_per_demographic_question("Education")

In [12]:
show_answers_per_demographic_question("Education", aggregate="Max")

In [13]:
show_answers_per_demographic_question("Employment")

In [14]:
show_answers_per_demographic_question("Employment", aggregate="Max")

In [15]:
show_answers_per_demographic_question("Marital status")

In [16]:
show_answers_per_demographic_question("Marital status", aggregate="Max")

In [17]:
show_answers_per_demographic_question("Residence")

In [18]:
show_answers_per_demographic_question("Residence", aggregate="Max")

# <a id='6'>Median violence percent per country</a>

In [19]:
country_df = pd.read_csv("https://github.com/DeborAguiar/BreakingSilence/blob/main/data/wikipedia-iso-country-codes.csv?raw=true", encoding='utf8', sep=",")
country_df.columns = ["Country", "A2C", "A3C", "Num","ISO"]
datac_df = data_df.merge(country_df)

In [20]:
df = datac_df.groupby(["Country", "A3C"])["Value"].agg(["median", "max", "min", "mean"]).reset_index()
df.columns = ["Country", "A3C", "Median", "Max", "Min", "Mean"]

In [21]:
def show_on_map(df, aggregate, title):
    fig = go.Figure(data=go.Choropleth(
        locations = df['A3C'],
        z = df[aggregate],
        text = df['Country'],
        colorscale = 'Rainbow',
        autocolorscale=False,
        reversescale=False,
        marker_line_color='darkgray',
        marker_line_width=0.5,
        colorbar_tickprefix = '%',
        colorbar_title = 'Violence %',
    ))

    fig.update_layout(
        title_text=title,
        geo=dict(
            showframe=False,
            showcoastlines=False,
            projection_type='equirectangular'
        ),
        annotations = [dict(
            x=0.55,
            y=0.1,
            xref='paper',
            yref='paper',
            text='Source: <a href="https://www.kaggle.com/andrewmvd/violence-against-women-and-girls">\
                Violence against Women and Girls</a>',
            showarrow = False
        )]
    )

    fig.show()

In [22]:
show_on_map(df, "Median", "Violence against Women and Girls - Country median level of violence[%]")

In [23]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="No education"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women with No education - Country Median level of violence [%]")

In [24]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="Higher"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women with Higher education - Country Median level of violence [%]")

In [25]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="Rural"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women in Rural Areas - Country Median level of violence [%]")

In [26]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="Urban"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women in Urban Areas - Country Median level of violence [%]")

In [27]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="15-24"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women age 15-24 - Country Median level of violence [%]")

In [28]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="25-34"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women age 25-34 - Country Median level of violence [%]")

In [29]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="35-49"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["median"]).reset_index()
df.columns = ["Country", "A3C", "Median"]
show_on_map(df, "Median", "Violence against Women age 35-49 - Country Median level of violence [%]")

# <a id='7'>Maximum violence percent per country</a>

In [30]:
df = datac_df.groupby(["Country", "A3C"])["Value"].agg(["median", "max", "min", "mean"]).reset_index()
df.columns = ["Country", "A3C", "Median", "Max", "Min", "Mean"]
show_on_map(df, "Max", "Violence against Women and Girls - Country maximum level of violence[%]")

In [31]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="No education"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["max"]).reset_index()
df.columns = ["Country", "A3C", "Max"]
show_on_map(df, "Max", "Violence against Women with No education - Country Max level of violence [%]")

In [32]:
filter_df = datac_df.loc[datac_df["Demographics Response"]=="Higher"]
df = filter_df.groupby(["Country", "A3C"])["Value"].agg(["max"]).reset_index()
df.columns = ["Country", "A3C", "Max"]
show_on_map(df, "Max", "Violence against Women with Higher education - Country Max level of violence [%]")

# <a id='8'>Summary of findings</a>


Some of the conclusions of the analysis are listed here:

* The most exposed age group is the 15-24, with a median of 17.50 and max of 80.1, while the absolute max is for 25-34 age group;

* Education level of girls and women is a very good predictor for violence against them: a higher education level will ensure a small median percent of 4.2%, while No education shows a 21.55% median and a 82% maximum violence percent.

* Employment factor also counts, from Unemployed to Employed for kind the median is varying from 14.55% to 20.15%. The women that work tend to be more exposed to violence.

* Marriage is slighly protecting if we look to the median, but married women are more exposed to higher level of violence. Widowed, divorced or separated have a smaller median than Married or living together group, as well as the max. But it might be also an age factor here.

* Women from Urban areas are less exposed to violence than women from Rural areas, both the median and the max values being smaller in Urban vs. Rural areas.

* Most exposed girls and women to extreme violence are from countries as Timor, Mali (subsaharian Africa), Chad, Eritreea, Afganistan and, surprisingly, Marroco. If the Marroco is a surprise, we should notice that the other countries are also either conflict zones or failes states, which is expected  to aggravate the life conditions of most exposed people.

* In the highest violence countries, the worst condition (i.e. exposure to higher levels of violence) is that of women or girls without education.



